lib/hutch/broker.rb

```diff
@@ -151,20 +151,24 @@ def bindings
       results
     end
 
+    # Find the existing bindings, and unbind any redundant bindings
+    def unbind_redundant_bindings(queue, routing_keys)
+      return unless http_api_use_enabled?
+
+      bindings.each do |dest, keys|
```

Comment:
Previously we iterated over bindings twice: once to filter, and a second time to do our work. Granted, the second time would likely iterate over a subset of bindings. Better still to iterate once if possible, skipping those we would've filtered and doing the necessary work otherwise.